src/ucp/proto/proto_init.c

```diff
@@ -648,18 +651,63 @@ ucp_proto_common_init_recv_perf(const ucp_proto_common_init_params_t *params,
     return UCS_OK;
 }
 
+static ucs_status_t
+ucp_proto_init_single_frag_ranges(const ucp_proto_common_init_params_t *params,
+                                  const ucp_proto_common_tl_perf_t *tl_perf,
+                                  ucp_proto_perf_node_t *const tl_perf_node,
+                                  ucp_md_map_t reg_md_map, size_t range_start,
+                                  size_t frag_size) {
+    int empty_msg = (frag_size == 0);
+    ucp_proto_perf_range_t xfer_perf, send_perf, recv_perf;
+    const ucp_proto_perf_range_t *parallel_stages[3];
+    ucs_status_t status;
+
+    if (empty_msg && (params->super.caps->min_length > 0)) {
+        return UCS_OK;
```

Comment:
when can it happen?

Reply:
Theoretically it can happen for atomic protocols (see `ucp_proto_amo_init`)

Reply:
maybe change to assert then? We do not expect 0-sized atomics

Reply:
get_zcopy can also have min_length>0

Reply:
then this protocol should not be considered for 0-size message range

Reply:
1. right, but the initialization is called for the whole protocol, not for a specific range
2. in UCT level, when scatter2cqe is enabled (by default), get_zcopy cannot be done for 0-64B message

Reply:
then we have to start range from min_length rather than from 1
discussed that with Ivan offline

Reply:
After the discussion with @brminich it was decided to move this if outside of this function.